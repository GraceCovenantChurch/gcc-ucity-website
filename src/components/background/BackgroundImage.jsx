import React from "react";
import PropTypes from "prop-types";

import styles from "./BackgroundImage.module.scss";

const BackgroundImage = (props) => {
  return (
    <div
      className={styles.background}
      style={{ backgroundImage: `url(${props.imageURL})` }}
    >
      <section className={styles.content}>{props.children}</section>
    </div>
  );
};

BackgroundImage.propTypes = {
  imageURL: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default BackgroundImage;
