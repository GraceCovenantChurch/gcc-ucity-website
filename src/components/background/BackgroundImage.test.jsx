import React from "react";
import renderer from "react-test-renderer";
import BackgroundImage from "./BackgroundImage";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BackgroundImage imageURL={"/something/someFilePath"}>
        {" "}
        <div />{" "}
      </BackgroundImage>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
