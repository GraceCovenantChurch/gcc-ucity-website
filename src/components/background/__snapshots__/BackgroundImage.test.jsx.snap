// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="background"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <section
    className="content"
  >
     
    <div />
     
  </section>
</div>
`;
